src/main/webapp/pages/pipelines/types/generic_pipeline.html

```diff
@@ -96,6 +97,23 @@ <h1 th:text="${mainTitle}"></h1>
 								</div>
 							</div>
 						</div>
+						<div th:if="${toolDataTableRequired}" class="form-group">
+							<div class="form-group" ng-repeat="toolDataTable in pipeline.toolDataTables.availableSettings">
```

Comment:
You have a nested `class="form-group"` here that is causing your alignment issues.  The select field also does not need to have a restricted width.

Can you re-write this section to be similar to:

```html
<div th:if="${toolDataTableRequired}" class="form-group"
        ng-repeat="toolDataTable in pipeline.toolDataTables.availableSettings">
    <label class="col-sm-2 control-label">{{toolDataTable.label}}</label>
    <div class="col-sm-10">
        <select id="{{toolDataTable.id}}" name="{{toolDataTable.label}}"
                     class="form-control"
                     ng-model="pipeline.selectedToolDataTableField"
                     ng-change="pipeline.toolDataTableFieldSelected('{{toolDataTable.id}}')"
                     ng-options="item.label for item in toolDataTable.parameters">
        </select>
    </div>
 </div>
```